llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -4099,23 +4153,33 @@ LoopVectorizationCostModel::computeMaxVF(ElementCount UserVF, unsigned UserIC) {
            "Invalid loop count");
     const SCEV *ExitCount = SE->getAddExpr(
         BackedgeTakenCount, SE->getOne(BackedgeTakenCount->getType()));
-    const SCEV *Rem = SE->getURemExpr(
-        SE->applyLoopGuards(ExitCount, TheLoop),
-        SE->getConstant(BackedgeTakenCount->getType(), MaxVFtimesIC));
-    if (Rem->isZero()) {
-      // Accept MaxFixedVF if we do not have a tail.
-      LLVM_DEBUG(dbgs() << "LV: No tail will remain for any chosen VF.\n");
-      return MaxFactors;
```

Comment:
Why change this to not exit early? No tail is needed but we still set the tail-folding style?